### DISCUSSION TITLE
Receiving Error message: MethodError: no method matching isnan(::Symbol)

### DISCUSSION THREAD

**Discussion by filipnykvist** [spine-tools/SpineOpt.jl#939] [Support]
Hi!

We're trying to simulate a river system and have imported nodes, units etc from an excel file. The input database looks similar to a working model from another case study. When we're trying to run our simulation we get the error-message below. We have tried updating both SpineOpt and julia.
What are we doing wrong? 

julia> # Running julia run_spineOpt.jl http://127.0.0.1:61054 http://127.0.0.1:61057
- [variable_node_state]2-element Vector{String}:
 "http://127.0.0.1:61054"
 "http://127.0.0.1:61057"

Execution started at 2024-03-13T12:37:15.015
Preparing SpineOpt for sqlite:///C:\Users\antes\OneDrive\Documents\KTH\År 3\KexJobb\Kexpriment_v1\.spinetoolbox\items\input\input.sqlite?spinedbfilter=cfg%3Atool%3Aobject_activity_control...
Initializing data structure from db...  0.914390 seconds (659.36 k allocations: 47.798 MiB, 44.03% compilation time)
Preprocessing data structure...  0.075105 seconds (157.15 k allocations: 11.652 MiB, 86.19% compilation time)
Checking data structure...  0.001157 seconds (13.58 k allocations: 1.283 MiB)
Running SpineOpt...
Creating temporal structure...  0.007103 seconds (100.07 k allocations: 6.410 MiB)
Creating stochastic structure...  0.000867 seconds (6.66 k allocations: 845.188 KiB)
Adding variables...
- [variable_units_available]  0.291812 seconds (1.77 M allocations: 214.672 MiB, 33.93% gc time)
- [variable_units_on]  0.747518 seconds (2.25 M allocations: 249.338 MiB, 68.64% gc time)
- [variable_units_started_up]  0.215696 seconds (1.75 M allocations: 210.816 MiB, 12.28% gc time)
- [variable_units_shut_down]  0.246733 seconds (1.75 M allocations: 211.877 MiB, 8.76% gc time)
- [variable_unit_flow]  0.785191 seconds (8.47 M allocations: 929.001 MiB, 12.41% gc time)
- [variable_unit_flow_op]  0.699616 seconds (4.89 M allocations: 609.881 MiB, 46.04% gc time, 0.67% compilation time)
- [variable_unit_flow_op_active]  0.413463 seconds (4.09 M allocations: 559.481 MiB, 9.28% gc time)
- [variable_connection_flow]  0.820990 seconds (8.35 M allocations: 991.116 MiB, 13.27% gc time, 1.85% compilation time)
- [variable_connection_intact_flow]  1.324016 seconds (8.14 M allocations: 980.012 MiB, 49.04% gc time)
- [variable_connections_invested]  0.000179 seconds (173 allocations: 20.188 KiB)
- [variable_connections_invested_available]  0.000112 seconds (178 allocations: 22.703 KiB)
- [variable_connections_decommissioned]  0.000064 seconds (114 allocations: 15.125 KiB)
- [variable_storages_invested]  0.000092 seconds (173 allocations: 20.188 KiB)
- [variable_storages_invested_available]  0.000073 seconds (172 allocations: 20.203 KiB)
- [variable_storages_decommissioned]  0.000045 seconds (114 allocations: 15.125 KiB)
ERROR: LoadError: MethodError: no method matching isnan(::Symbol)

Closest candidates are:
  isnan(::BigFloat)
   @ Base mpfr.jl:982
  isnan(::Missing)
   @ Base missing.jl:101
  isnan(::DualNumbers.Dual)
   @ DualNumbers C:\Users\antes\.julia\packages\DualNumbers\5knFX\src\dual.jl:52
  ...

Stacktrace:
  [1] _fix(var::JuMP.VariableRef, fix_value::Symbol)
    @ SpineInterface C:\Users\antes\.julia\packages\SpineInterface\uCIf7\src\update_model.jl:181
  [2] fix(var::JuMP.VariableRef, call::SpineInterface.Call)
    @ SpineInterface C:\Users\antes\.julia\packages\SpineInterface\uCIf7\src\update_model.jl:176
  [3] _variable(m::JuMP.Model, name::Symbol, ind::@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}, bin::Nothing, int::Nothing, lb::SpineInterface.Parameter, ub::Nothing, fix_value::SpineInterface.Parameter, internal_fix_value::Nothing, replacement_value::Nothing)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_common.jl:131
  [4] #861
    @ .\none:0 [inlined]
  [5] iterate
    @ .\generator.jl:47 [inlined]
  [6] _all(f::Base.var"#384#386", itr::Base.Generator{Vector{@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}}, SpineOpt.var"#861#862"{Nothing, Nothing, SpineInterface.Parameter, Nothing, SpineInterface.Parameter, Nothing, Nothing, JuMP.Model, Symbol}}, ::Colon)
    @ Base .\reduce.jl:1297
  [7] all
    @ .\reduce.jl:1283 [inlined]
  [8] Dict(kv::Base.Generator{Vector{@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}}, SpineOpt.var"#861#862"{Nothing, Nothing, SpineInterface.Parameter, Nothing, SpineInterface.Parameter, Nothing, Nothing, JuMP.Model, Symbol}})
    @ Base .\dict.jl:111
  [9] add_variable!(m::JuMP.Model, name::Symbol, indices::typeof(node_state_indices); bin::Nothing, int::Nothing, lb::SpineInterface.Parameter, ub::Nothing, initial_value::SpineInterface.Parameter, fix_value::SpineInterface.Parameter, internal_fix_value::Nothing, replacement_value::Nothing, non_anticipativity_time::Nothing, non_anticipativity_margin::Nothing)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_common.jl:56
 [10] add_variable_node_state!(m::JuMP.Model)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_node_state.jl:43
 [11] macro expansion
    @ .\timing.jl:279 [inlined]
 [12] macro expansion
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\util\misc.jl:51 [inlined]
 [13] _add_variables!(m::JuMP.Model; add_user_variables::SpineOpt.var"#44#54", log_level::Int64)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:122
 [14] macro expansion
    @ .\timing.jl:279 [inlined]
 [15] macro expansion
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\util\misc.jl:51 [inlined]
 [16] init_model!(m::JuMP.Model; add_user_variables::Function, add_constraints::Function, log_level::Int64)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:76
 [17] init_model!
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:75 [inlined]
 [18] rerun_spineopt_standard!(m::JuMP.Model, url_out::String; add_user_variables::Function, add_constraints::Function, log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!))
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:37
 [19] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{add_user_variables::SpineOpt.var"#44#54", add_constraints::SpineOpt.var"#45#55", log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!)})
    @ Base .\essentials.jl:894
 [20] rerun_spineopt(url_out::String; mip_solver::Nothing, lp_solver::Nothing, add_user_variables::Function, add_constraints::Function, log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, use_direct_model::Bool, run_kernel::typeof(run_spineopt_kernel!))
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:258
 [21] rerun_spineopt
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:235 [inlined]
 [22] _run_spineopt(url_in::String, url_out::String; upgrade::Bool, log_level::Int64, filters::Dict{String, String}, templates::Tuple{}, kwargs::@Kwargs{mip_solver::Nothing, lp_solver::Nothing, add_user_variables::SpineOpt.var"#44#54", add_constraints::SpineOpt.var"#45#55", optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, use_direct_model::Bool, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!)})
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:178
 [23] _run_spineopt
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:174 [inlined]
 [24] #run_spineopt#35
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:101 [inlined]
 [25] top-level scope
    @ C:\Users\antes\.spinetoolbox\work\spineopt_518dbc3d946543169caa5ce69180b44e_toolbox\run_spineOpt.jl:3
 [26] include(fname::String)
    @ Base.MainInclude .\client.jl:489
 [27] top-level scope
    @ none:1
in expression starting at C:\Users\antes\.spinetoolbox\work\spineopt_518dbc3d946543169caa5ce69180b44e_toolbox\run_spineOpt.jl:3

caused by: MethodError: no method matching isnan(::Symbol)

Closest candidates are:
  isnan(::BigFloat)
   @ Base mpfr.jl:982
  isnan(::Missing)
   @ Base missing.jl:101
  isnan(::DualNumbers.Dual)
   @ DualNumbers C:\Users\antes\.julia\packages\DualNumbers\5knFX\src\dual.jl:52
  ...

Stacktrace:
  [1] _fix(var::JuMP.VariableRef, fix_value::Symbol)
    @ SpineInterface C:\Users\antes\.julia\packages\SpineInterface\uCIf7\src\update_model.jl:181
  [2] fix(var::JuMP.VariableRef, call::SpineInterface.Call)
    @ SpineInterface C:\Users\antes\.julia\packages\SpineInterface\uCIf7\src\update_model.jl:176
  [3] _variable(m::JuMP.Model, name::Symbol, ind::@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}, bin::Nothing, int::Nothing, lb::SpineInterface.Parameter, ub::Nothing, fix_value::SpineInterface.Parameter, internal_fix_value::Nothing, replacement_value::Nothing)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_common.jl:131
  [4] #861
    @ .\none:0 [inlined]
  [5] iterate
    @ .\generator.jl:47 [inlined]
  [6] Dict{@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}, JuMP.VariableRef}(kv::Base.Generator{Vector{@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}}, SpineOpt.var"#861#862"{Nothing, Nothing, SpineInterface.Parameter, Nothing, SpineInterface.Parameter, Nothing, Nothing, JuMP.Model, Symbol}})
    @ Base .\dict.jl:85
  [7] dict_with_eltype
    @ .\abstractdict.jl:581 [inlined]
  [8] dict_with_eltype
    @ .\abstractdict.jl:588 [inlined]
  [9] Dict(kv::Base.Generator{Vector{@NamedTuple{node::SpineInterface.Object, stochastic_scenario::SpineInterface.Object, t::SpineInterface.TimeSlice}}, SpineOpt.var"#861#862"{Nothing, Nothing, SpineInterface.Parameter, Nothing, SpineInterface.Parameter, Nothing, Nothing, JuMP.Model, Symbol}})
    @ Base .\dict.jl:109
 [10] add_variable!(m::JuMP.Model, name::Symbol, indices::typeof(node_state_indices); bin::Nothing, int::Nothing, lb::SpineInterface.Parameter, ub::Nothing, initial_value::SpineInterface.Parameter, fix_value::SpineInterface.Parameter, internal_fix_value::Nothing, replacement_value::Nothing, non_anticipativity_time::Nothing, non_anticipativity_margin::Nothing)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_common.jl:56
 [11] add_variable_node_state!(m::JuMP.Model)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\variables\variable_node_state.jl:43
 [12] macro expansion
    @ .\timing.jl:279 [inlined]
 [13] macro expansion
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\util\misc.jl:51 [inlined]
 [14] _add_variables!(m::JuMP.Model; add_user_variables::SpineOpt.var"#44#54", log_level::Int64)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:122
 [15] macro expansion
    @ .\timing.jl:279 [inlined]
 [16] macro expansion
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\util\misc.jl:51 [inlined]
 [17] init_model!(m::JuMP.Model; add_user_variables::Function, add_constraints::Function, log_level::Int64)
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:76
 [18] init_model!
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:75 [inlined]
 [19] rerun_spineopt_standard!(m::JuMP.Model, url_out::String; add_user_variables::Function, add_constraints::Function, log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!))
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:37
 [20] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{add_user_variables::SpineOpt.var"#44#54", add_constraints::SpineOpt.var"#45#55", log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!)})
    @ Base .\essentials.jl:894
 [21] rerun_spineopt(url_out::String; mip_solver::Nothing, lp_solver::Nothing, add_user_variables::Function, add_constraints::Function, log_level::Int64, optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, resume_file_path::Nothing, use_direct_model::Bool, run_kernel::typeof(run_spineopt_kernel!))
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:258
 [22] rerun_spineopt
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:235 [inlined]
 [23] _run_spineopt(url_in::String, url_out::String; upgrade::Bool, log_level::Int64, filters::Dict{String, String}, templates::Tuple{}, kwargs::@Kwargs{mip_solver::Nothing, lp_solver::Nothing, add_user_variables::SpineOpt.var"#44#54", add_constraints::SpineOpt.var"#45#55", optimize::Bool, update_names::Bool, alternative::String, write_as_roll::Int64, use_direct_model::Bool, resume_file_path::Nothing, run_kernel::typeof(run_spineopt_kernel!)})
    @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:178
 [24] _run_spineopt
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:174 [inlined]
 [25] #run_spineopt#35
    @ C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt.jl:101 [inlined]
 [26] top-level scope
    @ C:\Users\antes\.spinetoolbox\work\spineopt_518dbc3d946543169caa5ce69180b44e_toolbox\run_spineOpt.jl:3
 [27] include(fname::String)
    @ Base.MainInclude .\client.jl:489
 [28] top-level scope
    @ none:1

**Comment by tarskul** [comment]
To exclude issues with the installation you could check whether you can run the [simple system](https://github.com/spine-tools/SpineOpt.jl/blob/master/examples/simple_system.json) (import the json file into a new spine db through the spine db editor).

You should also check whether your importer is up to date. You could export your spine db to a json file and compare that file to the current [template](https://github.com/spine-tools/SpineOpt.jl/blob/master/templates/spineopt_template.json). Are there any differences in definitions and default values?

**Comment by manuelma** [comment]
Looking at this

```
[1] _fix(var::JuMP.VariableRef, fix_value::Symbol)
```
it looks like it's trying to fix a variable to a Symbol. Most likely you have specified some initial or fix value for a variable using a string rather than a number.

**Comment by DillonJ** [comment]
This functionality would prevent this sort of error
https://github.com/spine-tools/Spine-Toolbox/issues/2176

The issue is tagged with toolbox milestone V0.8 - would be nice if it could get done

**Comment by filipnykvist** [comment]
Thank you for your help. We've found a few commas that we changed into dots instead in the excel sheet. Looks like the importer could read it but SpineOpt couldn't. Anyways, now SpineOpt can complete the entire run and we receive no errors. But we still don't receive an output report and there's a warning that says:
"using JuMP.parameter_value in module SpineInterface conflicts with an existing identifier."

Here's the full log:
julia> # Running julia run_spineOpt.jl http://127.0.0.1:61130 http://127.0.0.1:61133
2-element Vector{String}:
 "http://127.0.0.1:61130"
 "http://127.0.0.1:61133"

Execution started at 2024-03-13T15:33:12.534
Preparing SpineOpt for sqlite:///C:\Users\antes\OneDrive\Documents\KTH\År 3\KexJobb\Kexpriment_v1\.spinetoolbox\items\input\input.sqlite?spinedbfilter=cfg%3Atool%3Aobject_activity_control...
Initializing data structure from db...  0.565318 seconds (190.22 k allocations: 14.898 MiB)
Preprocessing data structure...  0.027620 seconds (89.54 k allocations: 7.207 MiB, 46.13% compilation time)
Checking data structure...  0.001248 seconds (13.58 k allocations: 1.283 MiB)
Running SpineOpt...
Creating temporal structure...  0.008943 seconds (100.07 k allocations: 6.411 MiB)
Creating stochastic structure...  0.001075 seconds (6.66 k allocations: 845.188 KiB)
Adding variables...
- [variable_units_available]  0.217543 seconds (1.77 M allocations: 214.907 MiB)
- [variable_units_on]  0.250827 seconds (2.25 M allocations: 249.574 MiB)
- [variable_units_started_up]  0.607540 seconds (1.75 M allocations: 211.051 MiB, 61.58% gc time)
- [variable_units_shut_down]  0.239529 seconds (1.75 M allocations: 212.112 MiB)
- [variable_unit_flow]  3.479550 seconds (8.47 M allocations: 920.029 MiB, 43.27% gc time)
- [variable_unit_flow_op]  0.421825 seconds (4.89 M allocations: 591.206 MiB, 11.63% gc time)
- [variable_unit_flow_op_active]  0.449225 seconds (4.09 M allocations: 544.413 MiB, 13.33% gc time)
- [variable_connection_flow]  0.479934 seconds (4.38 M allocations: 548.283 MiB, 11.69% gc time)
- [variable_connection_intact_flow]  0.467462 seconds (4.38 M allocations: 546.031 MiB, 11.51% gc time)
- [variable_connections_invested]  0.000253 seconds (173 allocations: 20.188 KiB)
- [variable_connections_invested_available]  0.000186 seconds (178 allocations: 22.703 KiB)
- [variable_connections_decommissioned]  0.000050 seconds (114 allocations: 15.125 KiB)
- [variable_storages_invested]  0.000152 seconds (173 allocations: 20.188 KiB)
- [variable_storages_invested_available]  0.000096 seconds (172 allocations: 20.203 KiB)
- [variable_storages_decommissioned]  0.000048 seconds (114 allocations: 15.125 KiB)
- [variable_node_state]  0.213737 seconds (2.24 M allocations: 238.998 MiB, 11.84% gc time)
- [variable_node_slack_pos]  0.000129 seconds (40 allocations: 18.156 KiB)
- [variable_node_slack_neg]  0.000029 seconds (40 allocations: 18.156 KiB)
- [variable_node_injection]  0.218568 seconds (2.38 M allocations: 313.623 MiB, 12.75% gc time)
- [variable_units_invested]  0.000128 seconds (173 allocations: 20.188 KiB)
- [variable_units_invested_available]  0.000108 seconds (172 allocations: 20.203 KiB)
- [variable_units_mothballed]  0.000047 seconds (114 allocations: 15.125 KiB)
- [variable_nonspin_units_started_up]  0.490968 seconds (3.87 M allocations: 573.933 MiB, 14.92% gc time)
- [variable_nonspin_units_shut_down]  0.452581 seconds (3.87 M allocations: 573.933 MiB, 8.86% gc time)
- [variable_node_pressure]  0.000249 seconds (469 allocations: 33.203 KiB)
- [variable_node_voltage_angle]  0.000134 seconds (158 allocations: 27.297 KiB)
- [variable_binary_gas_connection_flow]  0.000265 seconds (776 allocations: 99.891 KiB)
- [variable_user_constraint_slack_pos]  0.000118 seconds (34 allocations: 18.047 KiB)
- [variable_user_constraint_slack_neg]  0.000033 seconds (34 allocations: 18.047 KiB)
- [user_defined]  0.000000 seconds
  7.999994 seconds (46.09 M allocations: 5.604 GiB, 28.31% gc time)
Adding constraints...
- [constraint_unit_pw_heat_rate]  1.171673 seconds (14.53 M allocations: 872.289 MiB, 8.79% gc time)
- [constraint_user_constraint]  0.000040 seconds (28 allocations: 4.328 KiB)
- [constraint_node_injection]  3.904808 seconds (32.81 M allocations: 2.797 GiB, 7.98% gc time)
- [constraint_nodal_balance]  0.479226 seconds (4.90 M allocations: 561.471 MiB, 16.49% gc time)
- [constraint_candidate_connection_flow_ub]  0.000039 seconds (12 allocations: 2.094 KiB)
- [constraint_candidate_connection_flow_lb]  0.000079 seconds (261 allocations: 14.375 KiB)
- [constraint_connection_intact_flow_ptdf]  0.000027 seconds (15 allocations: 2.188 KiB)
- [constraint_connection_flow_intact_flow]  0.000034 seconds (15 allocations: 2.172 KiB)
- [constraint_connection_flow_lodf]  0.000048 seconds (23 allocations: 1.953 KiB)
- [constraint_connection_flow_capacity]  0.000111 seconds (158 allocations: 17.547 KiB)
- [constraint_connection_intact_flow_capacity]  0.000081 seconds (123 allocations: 7.875 KiB)
- [constraint_unit_flow_capacity]  8.006584 seconds (76.06 M allocations: 6.612 GiB, 9.73% gc time)
- [constraint_connections_invested_available]  0.000048 seconds (38 allocations: 2.531 KiB)
- [constraint_connection_lifetime]  0.000068 seconds (87 allocations: 6.016 KiB)
- [constraint_connections_invested_transition]  0.000064 seconds (26 allocations: 2.016 KiB)
- [constraint_storages_invested_available]  0.000038 seconds (38 allocations: 2.531 KiB)
- [constraint_storage_lifetime]  0.000070 seconds (89 allocations: 6.281 KiB)
- [constraint_storages_invested_transition]  0.000045 seconds (26 allocations: 2.016 KiB)
- [constraint_operating_point_bounds]  0.116972 seconds (1.33 M allocations: 167.226 MiB)
- [constraint_operating_point_rank]  0.166705 seconds (1.33 M allocations: 167.229 MiB, 21.09% gc time)
- [constraint_unit_flow_op_bounds]  1.232601 seconds (11.69 M allocations: 875.683 MiB, 8.70% gc time)
- [constraint_unit_flow_op_rank]  0.116306 seconds (1.33 M allocations: 168.615 MiB)
- [constraint_unit_flow_op_sum]  0.152833 seconds (820.58 k allocations: 96.966 MiB, 31.66% gc time)
- [constraint_fix_ratio_out_in_unit_flow]  1.469223 seconds (16.43 M allocations: 1.167 GiB, 7.02% gc time)
- [constraint_max_ratio_out_in_unit_flow]  0.000089 seconds (165 allocations: 18.250 KiB)
- [constraint_min_ratio_out_in_unit_flow]  0.000057 seconds (165 allocations: 18.250 KiB)
- [constraint_fix_ratio_out_out_unit_flow]  0.000055 seconds (165 allocations: 18.250 KiB)
- [constraint_max_ratio_out_out_unit_flow]  0.000049 seconds (165 allocations: 18.250 KiB)
- [constraint_min_ratio_out_out_unit_flow]  0.000034 seconds (165 allocations: 18.250 KiB)
- [constraint_fix_ratio_in_in_unit_flow]  0.000033 seconds (165 allocations: 18.250 KiB)
- [constraint_max_ratio_in_in_unit_flow]  0.000054 seconds (165 allocations: 18.250 KiB)
- [constraint_min_ratio_in_in_unit_flow]  0.000030 seconds (165 allocations: 18.250 KiB)
- [constraint_fix_ratio_in_out_unit_flow]  0.000046 seconds (165 allocations: 18.250 KiB)
- [constraint_max_ratio_in_out_unit_flow]  0.000027 seconds (165 allocations: 18.250 KiB)
- [constraint_min_ratio_in_out_unit_flow]  0.000064 seconds (165 allocations: 18.250 KiB)
- [constraint_ratio_out_in_connection_intact_flow]  0.000049 seconds (15 allocations: 2.156 KiB)
- [constraint_fix_ratio_out_in_connection_flow]  1.395487 seconds (16.28 M allocations: 1.135 GiB, 12.78% gc time, 13.12% compilation time)
- [constraint_max_ratio_out_in_connection_flow]  0.000067 seconds (158 allocations: 17.406 KiB)
- [constraint_min_ratio_out_in_connection_flow]  0.000059 seconds (158 allocations: 17.406 KiB)
- [constraint_node_state_capacity]  0.564068 seconds (5.19 M allocations: 479.968 MiB, 7.62% gc time)
- [constraint_cyclic_node_state]  0.000333 seconds (2.48 k allocations: 222.453 KiB)
- [constraint_max_total_cumulated_unit_flow_from_node]  0.000053 seconds (87 allocations: 9.484 KiB)
- [constraint_min_total_cumulated_unit_flow_from_node]  0.000028 seconds (90 allocations: 14.062 KiB)
- [constraint_max_total_cumulated_unit_flow_to_node]  0.000069 seconds (159 allocations: 17.359 KiB)
- [constraint_min_total_cumulated_unit_flow_to_node]  0.000048 seconds (159 allocations: 17.359 KiB)
- [constraint_units_on]  0.034786 seconds (392.79 k allocations: 52.491 MiB)
- [constraint_units_available]  0.601379 seconds (5.57 M allocations: 528.993 MiB, 14.50% gc time)
- [constraint_units_invested_available]  0.000040 seconds (38 allocations: 2.531 KiB)
- [constraint_unit_lifetime]  0.000035 seconds (53 allocations: 4.172 KiB)
- [constraint_units_invested_transition]  0.000045 seconds (26 allocations: 2.016 KiB)
- [constraint_minimum_operating_point]  0.049135 seconds (535.97 k allocations: 40.979 MiB)
- [constraint_min_down_time]  0.000032 seconds (54 allocations: 4.516 KiB)
- [constraint_min_up_time]  0.000032 seconds (52 allocations: 4.141 KiB)
- [constraint_unit_state_transition]  0.551707 seconds (4.76 M allocations: 599.121 MiB, 18.43% gc time)
- [constraint_ramp_up]  0.000342 seconds (466 allocations: 51.250 KiB)
- [constraint_ramp_down]  0.000109 seconds (466 allocations: 51.250 KiB)
- [constraint_non_spinning_reserves_lower_bound]  0.000630 seconds (2.55 k allocations: 286.438 KiB)
- [constraint_non_spinning_reserves_start_up_upper_bound]  0.000653 seconds (2.56 k allocations: 287.312 KiB)
- [constraint_non_spinning_reserves_shut_down_upper_bound]  0.000720 seconds (2.56 k allocations: 287.312 KiB)
- [constraint_res_minimum_node_state]  0.010581 seconds (2.90 k allocations: 218.102 KiB, 94.25% compilation time)
- [constraint_fix_node_pressure_point]  0.000109 seconds (160 allocations: 17.875 KiB)
- [constraint_connection_unitary_gas_flow]  0.000043 seconds (155 allocations: 16.984 KiB)
- [constraint_compression_ratio]  0.000032 seconds (157 allocations: 17.344 KiB)
- [constraint_storage_line_pack]  0.000033 seconds (157 allocations: 17.406 KiB)
- [constraint_connection_flow_gas_capacity]  0.000041 seconds (156 allocations: 17.203 KiB)
- [constraint_max_node_pressure]  0.000092 seconds (127 allocations: 8.734 KiB)
- [constraint_min_node_pressure]  0.000034 seconds (127 allocations: 8.734 KiB)
- [constraint_node_voltage_angle]  0.000031 seconds (88 allocations: 6.328 KiB)
- [constraint_max_node_voltage_angle]  0.000049 seconds (199 allocations: 13.391 KiB)
- [constraint_min_node_voltage_angle]  0.000035 seconds (199 allocations: 13.391 KiB)
- [constraint_investment_group_equal_investments]  0.000014 seconds (12 allocations: 1.500 KiB)
- [constraint_investment_group_minimum_entities_invested_available]  0.000005 seconds (8 allocations: 1.062 KiB)
- [constraint_investment_group_maximum_entities_invested_available]  0.000004 seconds (7 allocations: 992 bytes)
- [constraint_investment_group_minimum_capacity_invested_available]  0.000007 seconds (9 allocations: 1.156 KiB)
- [constraint_investment_group_maximum_capacity_invested_available]  0.000005 seconds (8 allocations: 1.062 KiB)
- [user_defined]  0.000000 seconds
 20.803448 seconds (203.42 M allocations: 16.931 GiB, 10.06% gc time, 0.93% compilation time)
┌ Warning: no objective terms defined
└ @ SpineOpt C:\Users\antes\.julia\packages\SpineOpt\FPoFr\src\run_spineopt_standard.jl:223
Setting objective...  0.003784 seconds (17.10 k allocations: 1.252 MiB)
Bringing model to the first window...  2.299255 seconds (24.60 M allocations: 2.743 GiB, 14.44% gc time)

Window 1: 2019-01-15T00:00~>2019-01-22T00:00
Optimizing model instance...Running HiGHS 1.6.0: Copyright (c) 2023 HiGHS under MIT licence terms
Presolving model
Problem status detected on presolve: Infeasible
Model   status      : Infeasible
Objective value     :  0.0000000000e+00
HiGHS run time      :          0.03
ERROR:   No LP invertible representation for getDualRay
  0.544018 seconds (1.26 M allocations: 121.099 MiB, 12.58% gc time)
model is infeasible - if conflicting constraints can be identified, they will be reported below
[ Info: The optimizer HiGHS.Optimizer does not support compute_conflict!
Writing report to sqlite:///C:\Users\antes\OneDrive\Documents\KTH\År 3\KexJobb\Kexpriment_v1\.spinetoolbox\items\output\output.sqlite?spinedbfilter=cfg%3Aexecution%3A%7B%22execution_item%22%3A+%22SpineOpt%22%2C+%22scenarios%22%3A+%5B%5D%2C+%22timestamp%22%3A+%222024-03-13T15%3A33%3A00%22%7D...  0.016332 seconds (442 allocations: 835.086 KiB)

Execution complete. Started at 2024-03-13T15:33:12.534, ended at 2024-03-13T15:33:44.829, elapsed time: 32 seconds, 295 milliseconds
A JuMP Model
Feasibility problem with:
Variables: 105732
JuMP.AffExpr-in-MathOptInterface.EqualTo{Float64}: 62328 constraints
JuMP.AffExpr-in-MathOptInterface.GreaterThan{Float64}: 168 constraints
JuMP.AffExpr-in-MathOptInterface.LessThan{Float64}: 43848 constraints
JuMP.VariableRef-in-MathOptInterface.EqualTo{Float64}: 111 constraints
JuMP.VariableRef-in-MathOptInterface.GreaterThan{Float64}: 92271 constraints
Model mode: AUTOMATIC
CachingOptimizer state: ATTACHED_OPTIMIZER
Solver name: HiGHS

**Comment by DillonJ** [comment]
Importer could read the values, but it read them as strings and that's what caused your error.

Regarding your output, did you specify any output items for reporting? See here: https://spine-tools.github.io/SpineOpt.jl/latest/getting_started/output_data/

**Reply by filipnykvist** [reply]
Yeah, we've specified the output.
It also says that: "Problem status detected on presolve: Infeasible"?
Could it be that our model is too large for SpineOpt to handle?

**Comment by DillonJ** [comment]
@filipnykvist it looks like this is a support issue so I have taken the liberty of converting it to a discussion in the support section.

**Reply by DillonJ** [reply]
Right, your model is infeasible. You will have to check all your nodes to make sure they can be balanced. You could consider using `node_slack_penalty` at some nodes to see where the problem is.

**Reply by filipnykvist** [reply]
Okej, i see. 
Might be a stupid question but, how do I know if the node Is balanced?

**Reply by DillonJ** [reply]
By default, nodes must be balanced where any outgoing flows must equal incoming flows less plus demand at that node. If you specify the node_slack_penalty, then this balance can be violated nut the violation is penalised in the objective function.

You just need to make sure that if any nodes have a demand, that there is something to meet that demand, or if there are incoming flows, there is somewhere for them to go, and vice versa.

Also, looks like you are using fix_or initial_ parameters - you would need to make sure these don't conflict with another constraint.
